app/models/menu.rb

```diff
@@ -1,5 +1,5 @@
 class Menu < ActiveRecord::Base
-  has_and_belongs_to_many :menu_categories
+  has_and_belongs_to_many :menu_items
```

Comment:
Using `uniq` only ensures that a menu has unique `menu_items`. The same applies for menu_items. I've added a database index in the join table to ensure that there are no duplicates.